Simplenote/src/main/java/com/automattic/simplenote/utils/TagUtils.java

```diff
@@ -102,11 +102,22 @@ public static boolean hashTagValid(String name) {
         try {
             String normalized = Normalizer.normalize(name, Normalizer.Form.NFC);
             String lowercased = normalized.toLowerCase(Locale.US);
-            String encoded = URLEncoder.encode(lowercased, StandardCharsets.UTF_8.name()).replace("*", "%2A").replace("+", "%20");
+            String encoded = replaceEncoded(URLEncoder.encode(lowercased, StandardCharsets.UTF_8.name()));
             return encoded.length() <= MAXIMUM_LENGTH_ENCODED_HASH;
         } catch (UnsupportedEncodingException e) {
             // TODO: Handle encoding exception with a custom UTF-8 encoder.
             return false;
         }
     }
+
+    /**
+     * Replace certain characters in @param encoded that were not encoded with encoded value.
+     *
+     * @param encoded   {@link String} to replace certain characters with encoded value.
+     *
+     * @return          {@link String} replaced characters with encoded values.
+     */
+    private static String replaceEncoded(String encoded) {
+        return encoded.replace("+", "%20").replace("*", "%2A").replace("-", "%2D").replace(".", "%2E").replace("_", "%5F");
```

Comment:
1. Minor (🔍 ): Consider formatting this line to multilines for better readability and for not having this line exceed a certain limit of character (ie. 120 character). Something like the below:
```
return encoded
                .replace("+", "%20")
                .replace("*", "%2A")
                .replace("-", "%2D")
                .replace(".", "%2E")
                .replace("_", "%5F");
```
2. Warning (⚠️ ): I understand that this is not part of this PR's change, but after looking it up I found that the encoded character for plus is the `%2B` instead of `%20`. Wdyt?

Reply:
1. There are no code style requirements like 120 characters limits in Simplenote.  There are a ton of lines longer than 120 characters.  Making the single line more readable by breaking it into multiple lines is subjective.  I updated the code style in 1525f768.

2. Actually, the `encoded` parameter has already been encoded upstream, which means plus characters are already `%2B` and space characters are `+`.  That's why we replace the `+` character in the `encoded` parameter with `%20` since it was a space character originally.  You can verify that by putting a breakpoint or log statement at the [`return encoded...` statement](https://github.com/Automattic/simplenote-android/pull/1218/files#diff-95498cc2ef506f4683acbb12c4a8e1af8cb36a0b6bd63656dec1a744a4b54507R121) and [adding `name = " +";` to the `try` statement in the `hashTagValid` method](https://github.com/Automattic/simplenote-android/pull/1218/files#diff-95498cc2ef506f4683acbb12c4a8e1af8cb36a0b6bd63656dec1a744a4b54507R102).  You'll see the `encoded` parameter will be `+%2B`.  I added details to the comment of the `replaceEncoded` method in 588dd4b8.

Reply:
1) Thanks for the change Tyler! I didn't know that there is no such requirement like 120 characters limits in Simplenote, good to know. Indeed all style related comments can be subjective, thus I am trying to use `Minor` with an emoji when commenting on those meaning that you can ignore as minor. Feel free to do so next time.
2) Thanks for the explanation Tyler! Also, thanks for actually included the reasoning behind it as a Javadoc.